linalg/ode.hpp

```diff
@@ -93,21 +132,19 @@ class ODESolver
    }
 
    /// Function for getting and setting the state vectors
-   virtual int   GetMaxStateSize() { return 0; }
-   virtual int   GetStateSize() { return 0; }
-   virtual const Vector &GetStateVector(int i)
-   {
-      mfem_error("ODESolver has no state vectors");
-      Vector *s = NULL; return *s; // Make some compiler happy
-   }
-   virtual void  GetStateVector(int i, Vector &state)
-   {
-      mfem_error("ODESolver has no state vectors");
-   }
-   virtual void  SetStateVector(int i, Vector &state)
-   {
-      mfem_error("ODESolver has no state vectors");
-   }
+   virtual int   GetMaxStateSize() { return state.GetMaxSize(); }
+   virtual int   GetStateSize() { return state.GetSize(); }
+   virtual const Vector &GetStateVector(int i) { return state.Get(i); }
+   virtual void  GetStateVector(int i, Vector &s) { state.Get(i,s); }
+   virtual void  SetStateVector(int i, Vector &s) { state.Set(i,s); }
+   virtual void  AddStateVector(Vector &s) { state.Add(s); }
+
+   static MFEM_EXPORT std::string ExplicitTypes;
+   static MFEM_EXPORT std::string ImplicitTypes;
+   static MFEM_EXPORT std::string Types;
+   static MFEM_EXPORT ODESolver *Select(const int ode_solver_type);
+   static MFEM_EXPORT ODESolver *SelectExplicit(const int ode_solver_type);
+   static MFEM_EXPORT ODESolver *SelectImplicit(const int ode_solver_type);
```

Comment:
Doxygen please :)

Reply:
I have added the const keywords, where I think they are appropriate.

Will work on the Doxygen, later this week.

Reply:
I have add /// comments.
Hope this is sufficient and clear.
Or are \** with or wo brief preferred?